--- a/drivers/mmc/sunxi_mmc.c
+++ b/drivers/mmc/sunxi_mmc.c
@@ -384,8 +384,14 @@
 		writel(data->blocks * data->blocksize, &priv->reg->bytecnt);
 	}
 
+#ifdef CONFIG_XPL_BUILD
+	/* Feed watchdog for Allwinner H5 */
+	writel((0xa57 << 1) | 1, (void *)0x01c20cb0);
+	writel((0xa57 << 1) | 1, (void *)0x01c20cc0);
+#else
 	debug("mmc %d, cmd %d(0x%08x), arg 0x%08x\n", priv->mmc_no,
 	      cmd->cmdidx, cmdval | cmd->cmdidx, cmd->cmdarg);
+#endif
 	writel(cmd->cmdarg, &priv->reg->arg);
 
 	if (!data)
@@ -611,6 +617,11 @@
 	ret = mmc_set_mod_clk(priv, 24000000);
 	if (ret)
 		return NULL;
+
+#if defined(CONFIG_XPL_BUILD)
+	if (sdc_no == 2)
+		cfg->b_max = 1;
+#endif
 
 	return mmc_create(cfg, priv);
 }
